### PR TITLE
Added legend onto map plots.

### PR DESCRIPTION
**Pull request by mollybsmith-noaa**
MATS component of pull request to add legends to scalar map plots.

### CODE REVIEW

File: apps/surface/server/dataFunctions/data_map.js [hunk 1]
```diff
@@ -155,19 +155,19 @@ dataMap = function (plotParams, plotFunction) {
     var cOptions = matsDataCurveOpsUtils.generateMapCurveOptions(curve, d, appParams, 0);  // generate map with site data
     dataset.push(cOptions);
 
-    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions("Lowest", dPurple);  // generate blue text layer
+    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions(matsTypes.ReservedWords.ScalarPurpleCurveText, dPurple);  // generate blue text layer
```
Comment (ian-noaa) [inline apps/surface/server/dataFunctions/data_map.js:158]: The comment here appears to be incorrect. 
```suggestion
    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions(matsTypes.ReservedWords.ScalarPurpleCurveText, dPurple);  // generate purple text layer
```

File: apps/surface/server/dataFunctions/data_map.js [hunk 2]
```diff
@@ -155,19 +155,19 @@ dataMap = function (plotParams, plotFunction) {
...
-    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions("Low", dBlue);  // generate blue text layer
+    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions(matsTypes.ReservedWords.ScalarBlueCurveText, dBlue);  // generate blue text layer
     dataset.push(cOptions);
 
-    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions("Moderate", dBlack);  // generate black text layer
+    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions(matsTypes.ReservedWords.ScalarBlackCurveText, dBlack);  // generate black text layer
     dataset.push(cOptions);
 
-    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions("High", dOrange);  // generate red text layer
+    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions(matsTypes.ReservedWords.ScalarOrangeCurveText, dOrange);  // generate red text layer
```
Comment (ian-noaa) [inline apps/surface/server/dataFunctions/data_map.js:167]: Ditto for this comment
```suggestion
    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions(matsTypes.ReservedWords.ScalarOrangeCurveText, dOrange);  // generate orange text layer
```

File: apps/upperair/server/dataFunctions/data_map.js [hunk 3]
```diff
@@ -169,19 +169,19 @@ dataMap = function (plotParams, plotFunction) {
     var cOptions = matsDataCurveOpsUtils.generateMapCurveOptions(curve, d, appParams, .0001);  // generate map with site data
     dataset.push(cOptions);
 
-    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions("Lowest", dPurple);  // generate blue text layer
+    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions(matsTypes.ReservedWords.ScalarPurpleCurveText, dPurple);  // generate blue text layer
```
Comment (ian-noaa) [inline apps/upperair/server/dataFunctions/data_map.js:172]:
```suggestion
    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions(matsTypes.ReservedWords.ScalarPurpleCurveText, dPurple);  // generate purple text layer
```

File: apps/upperair/server/dataFunctions/data_map.js [hunk 4]
```diff
@@ -169,19 +169,19 @@ dataMap = function (plotParams, plotFunction) {
...
-    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions("Low", dBlue);  // generate blue text layer
+    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions(matsTypes.ReservedWords.ScalarBlueCurveText, dBlue);  // generate blue text layer
     dataset.push(cOptions);
 
-    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions("Moderate", dBlack);  // generate black text layer
+    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions(matsTypes.ReservedWords.ScalarBlackCurveText, dBlack);  // generate black text layer
     dataset.push(cOptions);
 
-    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions("High", dOrange);  // generate red text layer
+    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions(matsTypes.ReservedWords.ScalarOrangeCurveText, dOrange);  // generate red text layer
```
Comment (ian-noaa) [inline apps/upperair/server/dataFunctions/data_map.js:181]:
```suggestion
    cOptions = matsDataCurveOpsUtils.generateMapColorTextOptions(matsTypes.ReservedWords.ScalarOrangeCurveText, dOrange);  // generate orange text layer
```

**Review by ian-noaa — APPROVED** [review]
Some small nits around comments that were incorrect. Otherwise, looks good!

File: apps/surface/server/dataFunctions/data_map.js (hunk 1 above)
Comment (mollybsmith-noaa) [reply apps/surface/server/dataFunctions/data_map.js:158]: Oops, thanks for catching the comments! That's what copy/paste does for a person...

**Pull request merged**